pacman/utilities/utility_objs/resource_tracker.py

```diff
@@ -1211,17 +1229,29 @@ def _available_resources(self, usable_chips):
         return n_cores, n_chips, max_sdram, n_tags
 
     def get_maximum_cores_available_on_a_chip(self):
-        """ returns the number of available cores of the chip with the maximum\
-        number of available cores
+        """ returns the number of available cores of a real chip with the \
+        maximum number of available cores
+
+        :return: the max cores available on the best real chip
+        :rtype: int
+        """
+        for n_cores_available, n_chips_with_n_cores in reversed(list(
+                enumerate(self._real_chips_with_n_cores_available))):
+            if n_chips_with_n_cores != 0:
+                return n_cores_available
 
-        :return: the max cores available on the best chip
+    def get_maximum_cores_available_on_a_virtual_chip(self):
```

Comment:
Most of this change is for this call - will you ever call this?  If not, the code just needs to keep the checks for virtual, but can miss out the actual count itself.

Reply:
as i mentioned in the commit message and pr:

It also allows users to ask for the vc with the most usable cores.......if this is usful........im not sure......... but as it comes out of the changes. it might as well be put in.

and given the code will only run on the vm chips, its not costing us much. I'm happy to take it out if your sure noone will ask. Just seemed a cheap bell (and yes i note what im saying to the standard software engineer)

Reply:
OK, happy enough then - it currently fails tests though...

Reply:
no idea what ya on about *cheeky smile* bloody f8